storm-core/src/jvm/org/apache/storm/blobstore/BlobStore.java

```diff
@@ -128,6 +128,15 @@
     public abstract void deleteBlob(String key, Subject who) throws AuthorizationException, KeyNotFoundException;
 
     /**
+     * Clear the blob data and metadata which is already removed from ZK.
+     * Implementation should just remove the blob as it is.
+     * This method should be used ONLY for syncing up blobs in Nimbus.
+     * @param key Key for the blob.
+     * @throws AuthorizationException
+     */
+    public abstract void clearRemovedBlob(String key) throws AuthorizationException;
```

Comment:
@HeartSaVioR :) May not want to assume no one implements their store when it is a public API. Anyways, good that  `deleteBlob` can be changed to handle this scenario.